### PR TITLE
Upgrade jupyter_server to version 1.19.1 to fix the vulnerability

### PR DESCRIPTION
**Pull request by HongW2019**

### CODE REVIEW

**Comment by HongW2019** [comment]
I tested this on host mode and it works. @jerrychenhf  Please have a review, thanks.

**Review by jerrychenhf — COMMENTED** [review]
We have many Spark related requirement.txt files which have the same contents. Can we share the same file to void manage the same version value across many files?

**Comment by HongW2019** [comment]
> We have many Spark related requirement.txt files which have the same contents. Can we share the same file to void manage the same version value across many files?

Have changed docker build context for optimized and native-sql Spark runtime to share the same file for Spark runtime.

**Comment by HongW2019** [comment]
@jerrychenhf Please have a review, thanks.

File: python/cloudtik/runtime/spark/scripts/services.sh [hunk 1]
```diff
@@ -21,7 +21,8 @@ start-head)
 stop-head)
     $HADOOP_HOME/bin/yarn --daemon stop resourcemanager
     $SPARK_HOME/sbin/stop-history-server.sh
-    jupyter lab stop
+    # workaround for stopping jupyter when password being set
+    kill $(pgrep jupyter)
```
Comment (jerrychenhf) [inline python/cloudtik/runtime/spark/scripts/services.sh:25]: If there is no jupyter server process, kill will missing pid parameter and show error:

kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]

Comment (HongW2019) [reply python/cloudtik/runtime/spark/scripts/services.sh]: Thanks for the review. Refined the scripts to stop running jupyter.

File: python/cloudtik/runtime/flink/scripts/services.sh [hunk 2]
```diff
@@ -18,14 +18,19 @@ start-head)
...
     nohup jupyter lab --no-browser > /tmp/logs/jupyterlab.log 2>&1 &
     ;;
 stop-head)
     $HADOOP_HOME/bin/yarn --daemon stop resourcemanager
     # Make sure HADOOP_CLASSPATH is set
     export HADOOP_CLASSPATH=`$HADOOP_HOME/bin/hadoop classpath`
     $FLINK_HOME/bin/historyserver.sh stop > /dev/null
-    jupyter lab stop
+    # workaround for stopping jupyter when password being set
+    if pgrep jupyter >/dev/null 2>&1; then
```
Comment (HongW2019) [inline python/cloudtik/runtime/flink/scripts/services.sh:30]: replace with variable

Comment (HongW2019) [reply python/cloudtik/runtime/flink/scripts/services.sh]: Done.

**Comment by HongW2019** [comment]
@jerrychenhf Please have a review, thanks.

**Pull request merged**